core/src/main/java/org/gbif/pipelines/core/parsers/legacy/Wgs84Projection.java

```diff
@@ -0,0 +1,150 @@
+      LOG.info("Wgs84Projection utils created");
+    } catch (FactoryRegistryException e) {
+      LOG.error("Failed to create geotools datum factory", e);
+    }
+  }
+
+  /**
+   * Reproject the given location into WGS84 location based on a known source datum or SRS.
+   * Darwin Core allows not only geodetic datums but also full spatial reference systems as values for "datum".
+   * The method will always return lat lons even if the processing failed. In that case only issues are set and the
+   * parsing result set to fail - but with a valid payload.
+   *
+   * @param lat   the original latitude
+   * @param lon   the original longitude
+   * @param datum the original geodetic datum the location are in
+   *
+   * @return the reprojected location or the original ones in case transformation failed
+   */
+  public static ParsedField<LatLng> reproject(double lat, double lon, String datum) {
+    Preconditions.checkArgument(lat >= -90d && lat <= 90d);
+    Preconditions.checkArgument(lon >= -180d && lon <= 180d);
+
+    List<InterpretationIssue> issues = new ArrayList<>();
+
+    if (Strings.isNullOrEmpty(datum)) {
+      issues.add(new InterpretationIssue(IssueType.GEODETIC_DATUM_ASSUMED_WGS84, DwcTerm.geodeticDatum));
+      return ParsedField.success(new LatLng(lat, lon), issues);
+    }
+
+    try {
+      CoordinateReferenceSystem crs = parseCRS(datum);
+      if (crs == null) {
+        issues.add(new InterpretationIssue(IssueType.GEODETIC_DATUM_INVALID, DwcTerm.geodeticDatum));
+        issues.add(new InterpretationIssue(IssueType.GEODETIC_DATUM_ASSUMED_WGS84, DwcTerm.geodeticDatum));
+
+      } else {
+        MathTransform transform = CRS.findMathTransform(crs, DefaultGeographicCRS.WGS84, true);
+        // different CRS may swap the x/y axis for lat lon, so check first:
+        double[] srcPt;
+        double[] dstPt = new double[3];
+        if (CRS.getAxisOrder(crs) == CRS.AxisOrder.NORTH_EAST) {
+          // lat lon
+          srcPt = new double[] {lat, lon, 0};
+        } else {
+          // lon lat
+          LOG.debug("Use lon/lat ordering for reprojection with datum={} and lat/lon={}/{}", datum, lat, lon);
+          srcPt = new double[] {lon, lat, 0};
+        }
+
+        transform.transform(srcPt, 0, dstPt, 0, 1);
+
+        double lat2 = dstPt[1];
+        double lon2 = dstPt[0];
+        // verify the datum shift is reasonable
+        if (Math.abs(lat - lat2) > SUSPICIOUS_SHIFT || Math.abs(lon - lon2) > SUSPICIOUS_SHIFT) {
+          issues.add(new InterpretationIssue(IssueType.COORDINATE_REPROJECTION_SUSPICIOUS, DwcTerm.geodeticDatum));
+          LOG.debug("Found suspicious shift for datum={} and lat/lon={}/{} so returning failure and keeping orig coord",
+            datum, lat, lon);
+          return ParsedField.fail(new LatLng(lat, lon), issues);
+        }
+        // flag the record if coords actually changed
+        if (lat != lat2 || lon != lon2) {
```

Comment:
this must be changed to (Double.compare(lat, lat2) + Double.compare(lon, lon2)) != 0 or something similar, compare 2 doubles in Java is vert unsafe: https://www.krenger.ch/blog/java-comparing-floating-point-numbers/